### ISSUE TITLE
Proper team balancing

### ISSUE THREAD

**Issue by deferred-impact** [deferred-impact/battledudes-issue-tracker#160]
This game **sucks** at balancing teams. It insists on having both teams equal in number, completely disregarding the fact that, even in this game, a single good player can handle multiple noobs.

There are 2 parts to this equation.

**First**, we need some way of assessing if teams are balanced. This is the hard part, and this question is something even big companies struggle with. Nevertheless, we cannot avoid it by declaring it a futile effort. Even a terrible team balancing algo would still be better than the dumb "if two teams have the same number of players, they balanced" that we currently have.

The naive approach is to use KDR. It would work, although KDR is an easily manipulated stat. KDR also does not necessarily represent a given player's skill - while i die a lot, and have very limited firepower with my Flintlock, i will nevertheless have little trouble dispatching 3 AR noobs. KPG suffers from the same problem. Still, as outlined above, even this would be better than whatever we have right now.

Accuracy might be a better stat to track. How much damage did the player actually deal (i.e. hit the enemy), vs how much damage they fired. While some weapons care about accuracy more than others, and while there are many other factors beyond accuracy (dodging, tactics, loadout selection etc.), good accuracy *tends* to correlate with good skills in general.

But by all means, if you have a better idea in mind, please suggest it. There are many possible solutions, and accuracy is far far far from ideal. In the extreme, you could even rig some neural network to this problem.

**Second**, how do we actually apply the rebalance after the algo decided on the option. This question is... easier, though still a bit complex. Two observations:

 - if you rebalance both teams completely, you are basically restarting the round from scratch. You may as well set the score back to 0. No, ideally rebalances would be minor, such as swapping two players.
 - if you force the rebalance upon people, they will understandably get pissed. Some players simply don't care about balance, and it is their right to do so. Some players would love to switch, but like their team/like some enemy that they're fighting/care about winning more than about having a good time (lmao)

This more or less forces us into the following approach:

The game picks 2 players to swap (using the algo outlined above; preferably the difference in skill between said players is as large as possible), and then displays a message to both such players. "Get 5 gems by swapping teams. Press ESC now to swap". Pressing ESC will open the usual settings menu, with a huge "swap teams" button displayed. Pressing the button will display "Confirmed. Please wait for the other player".

Once both players confirm the swap, it is done, and the players are rewarded with the whatever amount of gems was promised to them.
